servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -5026,6 +5077,15 @@ public void putPolicy(ResourceContext ctx, String domainName, String policyName,
 
         validatePolicyAssertions(policy.getAssertions(), caller);
 
+        // we are not going to allow any user to add/update a policy
+        // that one of its assertions has role associated that does not exist
+
+        RoleList roleList = getRoleList(ctx, domainName, null, null);
+        if (roleList != null && roleList.getNames() != null) {
+            String finalDomainName = domainName;
+            policy.getAssertions().forEach(assertion -> validateRoleAssociatedIsExist(new HashSet<>((roleList.getNames())), assertion.getRole(), finalDomainName, policy.getName(), caller));
```

Comment:
policy.getAssertions() I believe could be null so we must check here before.

Take a look at implementing this functionality within validatePolicyAssertions/validatePolicyAssertion calls.